src/cluster/eks_update_cluster_version.go

```diff
@@ -0,0 +1,117 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cluster
+
+import (
+	"time"
+
+	"emperror.dev/errors"
+	"go.uber.org/cadence"
+	"go.uber.org/cadence/workflow"
+
+	eksWorkflow "github.com/banzaicloud/pipeline/internal/cluster/distribution/eks/eksprovider/workflow"
+	pkgCadence "github.com/banzaicloud/pipeline/pkg/cadence"
+	pkgCluster "github.com/banzaicloud/pipeline/pkg/cluster"
+	pkgAmazon "github.com/banzaicloud/pipeline/pkg/providers/amazon"
+)
+
+const EKSUpdateClusterVersionWorkflowName = "eks-update-cluster-version"
+
+// EKSUpdateClusterVersionWorkflowInput holds data needed to update EKS cluster version
+type EKSUpdateClusterVersionWorkflowInput struct {
+	Region         string
+	OrganizationID uint
+	SecretID       string
+	ConfigSecretID string
+
+	ClusterID   uint
+	ClusterUID  string
+	ClusterName string
+
+	Version string
+}
+
+// EKSUpdateClusterVersionWorkflow executes the Cadence workflow responsible for updating EKS cluster version
+func EKSUpdateClusterVersionWorkflow(ctx workflow.Context, input EKSUpdateClusterVersionWorkflowInput) error {
+	ao := workflow.ActivityOptions{
+		ScheduleToStartTimeout: 10 * time.Minute,
+		StartToCloseTimeout:    5 * time.Minute,
+		WaitForCancellation:    true,
+		RetryPolicy: &cadence.RetryPolicy{
+			InitialInterval:          2 * time.Second,
+			BackoffCoefficient:       1.5,
+			MaximumInterval:          30 * time.Second,
+			MaximumAttempts:          5,
+			NonRetriableErrorReasons: []string{"cadenceInternal:Panic", eksWorkflow.ErrReasonStackFailed},
+		},
+	}
+
+	commonActivityInput := eksWorkflow.EKSActivityInput{
+		OrganizationID:            input.OrganizationID,
+		SecretID:                  input.SecretID,
+		Region:                    input.Region,
+		ClusterName:               input.ClusterName,
+		AWSClientRequestTokenBase: pkgAmazon.NewNormalizedClientRequestToken(workflow.GetInfo(ctx).WorkflowExecution.ID),
+	}
+
+	// execute the version update
+	var updateOutput eksWorkflow.UpdateClusterVersionActivityOutput
+	{
+		activityInput := eksWorkflow.UpdateClusterVersionActivityInput{
+			EKSActivityInput: commonActivityInput,
+			Version:          input.Version,
+		}
+		err := workflow.ExecuteActivity(ctx, eksWorkflow.UpdateClusterVersionActivityName, activityInput).Get(ctx, &updateOutput)
+		if err != nil {
+			err = errors.WrapIff(pkgCadence.UnwrapError(err), "%q activity failed", eksWorkflow.UpdateClusterVersionActivityName)
+			eksWorkflow.SetClusterStatus(ctx, input.ClusterID, pkgCluster.Warning, err.Error()) // nolint: errcheck
```

Comment:
```suggestion
			_ = eksWorkflow.SetClusterStatus(ctx, input.ClusterID, pkgCluster.Warning, err)
```

**Required**: we **MUST** unwrap the wrapped error here otherwise the cluster status message might end up being short without details such as `CLOUDFORMATION_STACK_FAILED` only, due to our Cadence error wrapping mechanism.
The `eksWorkflow.SetClusterStatus` function does the unwrapping, but it requires the original error, not just the error string for the details to be transferred.

Same at
https://github.com/banzaicloud/pipeline/pull/3123/files#diff-170c8394b68e0b56c2acd006e859f9c9R97
https://github.com/banzaicloud/pipeline/pull/3123/files#diff-170c8394b68e0b56c2acd006e859f9c9R110

Reply:
Thanks, fixed!